File: react-java/backend/src/main/java/com/clt/hrm/core/administration/role/dtos/RoleDto.java
package com.clt.hrm.core.administration.role.dtos;

import java.util.List;

import com.clt.hrm.infra.common.dtos.BaseDto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class RoleDto extends BaseDto {
    /**
	 * 
	 */
	private static final long serialVersionUID = -6288035537005720485L;
	private String roleId;
    private String roleCd;
    private String roleNm;
    private String roleDesc;
    List<RoleAuthDto> roleAuthList;
}

File: nestjs/backend/src/apps/administration/roles/dto/role.dto.ts
import { Type } from 'class-transformer';
import { IsArray, IsOptional, IsString, ValidateNested } from 'class-validator';
import { RoleAuthDto } from './role-auth.dto';
import { BaseDto } from '@infra/common/dtos';

export class RoleDto extends BaseDto {
  @IsOptional()
  @IsString()
  roleId?: string;

  @IsOptional()
  @IsString()
  roleCd?: string;

  @IsOptional()
  @IsString()
  roleNm?: string;

  @IsOptional()
  @IsString()
  roleDesc?: string;

  @IsOptional()
  @IsArray()
  @ValidateNested({ each: true })
  @Type(() => RoleAuthDto)
  roleAuthList?: RoleAuthDto[];
}
